code/modules/events/tear.dm

```diff
@@ -18,8 +18,10 @@
 	impact_area = findEventArea()
 
 /datum/event/tear/start()
-	var/turf/T = pick(get_area_turfs(impact_area))
-	if(T)
+	for(var/turf/T in get_area_turfs(impact_area))
+		if(is_blocked_turf(T))
+			continue
```

Comment:
Doesn't this mean it will always pick the same turf? Maybe a shuffling of the list first is needed?

Reply:
do you know offhand if there's a helper proc to do this? i feel like getting a random non-dense tile in a given area is a pretty common need but i couldn't find any utility methods to do it.

Reply:
PTAL